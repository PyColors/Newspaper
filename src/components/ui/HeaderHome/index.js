import { generator } from 'uigradients';
import styled from 'styled-components';

const HeaderHome = styled.header`
  ${generator({ gradient: 'electric_violet', type: 'radial' })} height: 180px;
  padding: 20px;
  color: white;
`;

export default HeaderHome;
