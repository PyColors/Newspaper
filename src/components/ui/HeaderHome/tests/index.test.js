import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import HeaderHome from '../index';

describe('<HeaderHome />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer
      .create(
        <header class="className">
          <h1 class="className">
            <span>Description</span>
          </h1>
          >
        </header>
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render an <HeaderHome> tag', () => {
    const renderedComponent = shallow(<HeaderHome />);
    expect(renderedComponent.type()).toEqual('header');
  });

  it('should render its text', () => {
    const children = 'Text';
    const renderedComponent = shallow(<HeaderHome>{children}</HeaderHome>);
    expect(renderedComponent.contains(children)).toBe(true);
  });

  it('should render a prop', () => {
    const id = 'testH1Id';
    const renderedComponent = shallow(<HeaderHome id={id} />);
    expect(renderedComponent.prop('id')).toEqual(id);
  });
});
