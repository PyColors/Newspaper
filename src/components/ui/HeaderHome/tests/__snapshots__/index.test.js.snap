// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderHome /> should match the snapshot 1`] = `
<header
  class="className"
>
  <h1
    class="className"
  >
    <span>
      Description
    </span>
  </h1>
  &gt;
</header>
`;
